### ISSUE TITLE
Add whatever it is console verifiers need to the SNES core

### ISSUE THREAD

**Issue by nattthebear** [TASEmulators/BizHawk#2010]
Bizhawk unsets the lag flag for the current frame if any of the following are observed:

* Cpu reads from `$4016`
* Cpu writes to `$4016:$4017`
* Cpu reads from `$4218:$421f`

This is great for TASers because it tells them whenever their input could possibly be perceived by the emulated game.

Some external tools would like a flag to tell them whenever the controller hardware was actually polled, which would presumably include only these situations:

* Cpu reads from `$4016`
* Cpu writes to `$4016:$4017`
* Automatic joypad poll occurs after vblank (as initiated by a previous write to `$4200`)

Getting the data is easy, but how do we expose it?

**Comment by YoshiRulz** [comment]
From #tasbot IRC/Discord. I don't think there's any novel info in here, but it gives a good overview of the problem.

> &lt;YoshiRulz> Because it's relevant again: can someone ELI5 why BizHawk BSNES dumps don't sync?
> &lt;TheMas3212> bizhawk's sence of when the game latches doesnt line up with when the console actually latches
> &lt;TheMas3212> 'lag frame'
> &lt;YoshiRulz> As in TAStudio sometimes shows green when there's no poll, or sometimes shows red when there is a poll, or both?
> &lt;TheMas3212> not sure about TAStudio but the issue becomes apparent in lua scripts using emu.islagged()
> &lt;YoshiRulz> They're the same. I just need to know if BizHawk dumps have more or less latches than lsnes.
> &lt;YoshiRulz> Or is it something more subtle?
> &lt;TheMas3212> i think it might be more subtle, but in general less?
> &lt;YoshiRulz> Hmm. My money's on "dumb typo in bridging code" then.
> &lt;dwangoAC> YoshiRulz: We'd definitely like to work out what the problem is as it's about to become a whole lot more substantial and visible - if this process of nudging the APU in the direction of more accuracy is successful it will mean there will be far more pressure on trying to get runs from BizHawk to dump cleanly.
> &lt;Ilari> YoshiRulz: If you want basic sanity check, check whatever you have does not count any lagged frames in-level in Super Mario World.
> &lt;Ilari> YoshiRulz: especially in places where SMW is known to lag (I think first such spot is in Yoshi's Island 4).
> &lt;Ilari> YoshiRulz: At minimum, to dump SNES correctly, script would need to know if controllers were latched during frame or not. If the frame was lag or not is not relevant.

**Comment by YoshiRulz** [comment]
Working on branch [`bsnes-lagfix-experiments`](https://github.com/TASVideos/BizHawk/tree/bsnes-lagfix-experiments). HMU on Discord if you have a replay device and are willing to dump and attempt playback.

**Comment by alyosha-tas** [comment]
I think this is resolved with the updates to the v115 core. Input latches are now correctly captured (accessiblethrough input callbacks and the event.oninputpoll in lua) and I even verified a TAS with inputs dumped in this way.
